### PR TITLE
temporal raster mean plot

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[Temporal Area Mean Plot](https://trello.com/c/T1f6sfmr/330-temporal-area-mean-plot)

File: projects/wave-core-new/src/lib/operators/dialogs/operator-list/operator-list.component.ts [hunk 1]
```diff
@@ -128,11 +128,14 @@ export class OperatorListComponent implements OnInit {
...
         },
-        //     {
-        //         component: TimePlotComponent,
-        //         type: TimePlotType,
-        //         description: 'Plot time data'
-        //     }
+        {
+            component: TemporalRasterMeanPlotDialogComponent,
+            type: {
+                NAME: 'Temporal Raster Mean Plot',
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/operators/dialogs/operator-list/operator-list.component.ts:134]: cf. my name dicsussion in the backend PR

File: projects/wave-core-new/src/lib/operators/dialogs/operator-list/operator-list.component.ts [hunk 2]
```diff
@@ -128,11 +128,14 @@ export class OperatorListComponent implements OnInit {
...
+                ICON_URL: createIconDataUrl('Temporal Raster Mean Plot'),
+            },
+            description: 'Create an area chart over means of raster time series'
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/operators/dialogs/operator-list/operator-list.component.ts:137]: "Create an area chart over the mean pixel values of the images of a raster time series"

File: projects/wave-core-new/src/lib/operators/dialogs/temporal-raster-mean-plot-dialog/temporal-raster-mean-plot-dialog.component.html [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<wave-sidenav-header>Temporal Raster Mean Plot</wave-sidenav-header>
+
+<wave-dialog-help>
+    <p>
+        This operator generates a <i>Temporal Raster Mean Plot</i> based on a raster data set (<i>Input</i>).
+        Select the <i>Time Tick Position</i>, which specifies the position of the x-axis ticks,
+        e.g. at the the time start, end or center of the raster's validity.
+        The plot shows an area chart of the mean values (y-axis) of the rasters.
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/operators/dialogs/temporal-raster-mean-plot-dialog/temporal-raster-mean-plot-dialog.component.html:8]: "The plot shows an area chart of the mean pixel values (y-axis) of the rasters."

Comment (michaelmattig) [reply projects/wave-core-new/src/lib/operators/dialogs/temporal-raster-mean-plot-dialog/temporal-raster-mean-plot-dialog.component.html:8]: or is it clear anyway?

File: projects/wave-core-new/src/lib/operators/dialogs/temporal-raster-mean-plot-dialog/temporal-raster-mean-plot-dialog.component.html [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        The x-axis displays the timestamps.
+        Edit the <i>Output Name</i> of the plot, if desired, and click on <i>Create</i> to generate the plot.
+    </p>
+    <p>
+        Note: Plots change dynamically with the map view and include only pixels or features, which are located in this area.
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/operators/dialogs/temporal-raster-mean-plot-dialog/temporal-raster-mean-plot-dialog.component.html:13]: maybe add that the plot only makes sense when selecting a time range?

Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/operators/dialogs/temporal-raster-mean-plot-dialog/temporal-raster-mean-plot-dialog.component.html]: 3328cae

File: projects/wave-core-new/src/lib/operators/dialogs/temporal-raster-mean-plot-dialog/temporal-raster-mean-plot-dialog.component.html (hunk 3 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/operators/dialogs/temporal-raster-mean-plot-dialog/temporal-raster-mean-plot-dialog.component.html]: 3328cae

File: projects/wave-core-new/src/lib/operators/dialogs/operator-list/operator-list.component.ts (hunk 2 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/operators/dialogs/operator-list/operator-list.component.ts]: 3328cae

File: projects/wave-core-new/src/lib/operators/dialogs/operator-list/operator-list.component.ts (hunk 1 above)
Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/operators/dialogs/operator-list/operator-list.component.ts:134]: 20d4db5

File: projects/wave-core-new/src/lib/operators/dialogs/mean-raster-pixel-values-over-time-dialog/mean-raster-pixel-values-over-time-dialog.component.html [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        <wave-layer-selection [types]="inputTypes" formControlName="layer"></wave-layer-selection>
+        <wave-dialog-section-heading title="Config" subtitle="Specify the Histogram"></wave-dialog-section-heading>
+        <mat-form-field>
+            <mat-select placeholder="Time Tick Position" formControlName="timePosition">
+                <mat-option *ngFor="let timePosition of timePositionOptions"
+                            [value]="timePosition.value">{{timePosition.label}}</mat-option>
+            </mat-select>
+        </mat-form-field>
+        <mat-form-field>
+            <mat-select placeholder="Fill area under curve" formControlName="area">
```
Comment (michaelmattig) [inline projects/wave-core-new/src/lib/operators/dialogs/mean-raster-pixel-values-over-time-dialog/mean-raster-pixel-values-over-time-dialog.component.html:31]: wouldn't a checkbox be more apropriate for a binary choice?

Comment (ChristianBeilschmidt) [reply projects/wave-core-new/src/lib/operators/dialogs/mean-raster-pixel-values-over-time-dialog/mean-raster-pixel-values-over-time-dialog.component.html]: 2cbf586

**Pull request merged**